server/athenian/api/controllers/selector.py

```diff
@@ -0,0 +1,80 @@
+from typing import Callable, Optional
+
+import aiomcache
+from sqlalchemy import and_, select
+
+from athenian.api.controllers.features.entries import get_calculator
+from athenian.api.models.state.models import AccountFeature, Feature, \
+    FeatureComponent, God
+from athenian.api.typing_utils import DatabaseLike
+
+
+async def get_calculator_for_user(
+    service: str,
+    account_id: int,
+    user_id: str,
+    sdb: DatabaseLike,
+    mdb: DatabaseLike,
+    pdb: DatabaseLike,
+    rdb: DatabaseLike,
+    cache: Optional[aiomcache.Client],
+    raise_err: Optional[bool] = False,
+    base_module: Optional[str] = "athenian.api.experiments",
+) -> Callable:
+    """Get the metrics calculator function for the given user."""
+
+    def _get_calculator(variation=None):
+        return get_calculator(
+            service, mdb, pdb, rdb, cache, variation=variation, raise_err=raise_err,
+            base_module=base_module,
+        )
+
+    feature_name_prefix = METRIC_ENTRIES_VARIATIONS_PREFIX[service]
+    all_metrics_variations_features = await sdb.fetch_all(
+        select([Feature.id, Feature.name, Feature.default_parameters]).where(
+            and_(
+                Feature.name.like(f"{feature_name_prefix}%"),
+                Feature.component == FeatureComponent.server,
+                Feature.enabled,
+            ),
+        ),
+    )
+
+    if not all_metrics_variations_features:
+        return _get_calculator()
+
+    all_metrics_variations_features = {
+        row[0]: {"name": row[1][len(feature_name_prefix):], "params": row[2] or {}}
```

Comment:
I found that taking values by index quickly goes out of sync with the selected columns list.
```suggestion
        row[Feature.id.key]: {"name": row[Feature.name.key][len(feature_name_prefix):], "params": row[Feature.default_parameters.key] or {}}
```

Reply:
also much more readable 👍